dmscripts/notify_suppliers_with_incomplete_applications.py

```diff
@@ -61,22 +74,14 @@ def notify_suppliers_with_incomplete_applications(framework_slug, stage, notify_
     error_count = 0
 
     for sf in data_api_client.find_framework_suppliers_iter(framework_slug):
-        message = ''
-        if not sf.get('applicationCompanyDetailsConfirmed', None):
-            message += MESSAGES['unconfirmed_company_details']
-        if not sf.get('declaration', {'status': None}).get('status', None) == 'complete':
-            message += MESSAGES['incomplete_declaration']
+        # Restrict suppliers to those specified in the argument, if given.
+        # While this is inefficient for a small number of supplier IDs, looking up
+        # each supplier individually for a large number of supplier IDs would be worse.
+        if supplier_ids:
+            if sf['supplierId'] not in supplier_ids:
+                continue
```

Comment:
could this be expressed using `filter` instead?

Reply:
Possibly? I doubt `filter` will gain us anything much performance-wise though. Also we'd have to do some fiddling to get it to work on this iterable, as the client method is doing some magic to paginate through our API's response (hence the `_iter` suffix).